generic-harvester/harvester/store/database_store.py

```diff
@@ -0,0 +1,141 @@
+"""
+
+"""
+
+import psycopg2
+import sqlalchemy as sa
+
+
+def format_dict(dictionary):
+    return ", ".join(["{}='{}'".format(key, value) for key, value in dictionary.items()])
+
+
+def reduce_space(value):
+    return ' '.join(value.split())
+
+
+# TODO: make this a context manager?
+
+class PostgresStore(object):
+
+    def __init__(self, db_params, logger):
+        self.url = db_params["driver"]+"://"+db_params["user"]+":"+db_params["password"]+"@"+db_params["host"]\
+            + ":"+db_params["port"]+"/"+db_params["database"]
+        self.logger = logger
+        self.engine = sa.create_engine(self.url)
+        self.conn = self.engine.connect()
+        self.trans = self.conn.begin()
+
+    def delete(self, table_name, key):
+        """
+        Delete records from table name where they match the provided key
+
+        :param table_name: name of table
+        :param key: a dictionary of column name/value pairs to use as the key
+        """
+
+        self.logger.info("Deleting records from {} where {}...".format(table_name, format_dict(key)))
+
+        condition = " ".join([field_name + " = :" + field_name for field_name in key])
```

Comment:
Don't need to turn it into a list here, join will just take the generator expression directly (i.e. identical, just without the square brackets).